.eslintrc.js

```diff
@@ -53,24 +46,21 @@ module.exports = {
       },
 
       rules: {
-        // the TypeScript compiler already takes care of this and
-        // leaving it enabled results in false positives for interface imports
-        'no-unused-vars': 'off',
+        '@typescript-eslint/ban-ts-comment': 'warn',
+        '@typescript-eslint/ban-types': 'warn',
```

Comment:
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/ban-types.md

I noticed a lot of usages of `any`, `object`, `Function`. This rule has a few options, but unsure how helpful they'll be here..

Reply:
We use `object` to mean "valid WeakMap key" in places, which I do think is a somewhat valid usage. Maybe we can make a type alias specifically for that, or start using record or some other type of key.

I'd like to try to slowly eliminate `any` and `Function` usage.